fxa-rust-client/src/errors.rs

```diff
@@ -133,8 +133,8 @@ pub enum ErrorKind {
     #[fail(display = "HAWK error: {}", _0)]
     HawkError(#[fail(cause)] SyncFailure<hawk::Error>),
 
-    #[fail(display = "JOSE error: {}", _0)]
-    JoseError(#[fail(cause)] SyncFailure<jose::error::Error>),
+    #[fail(display = "Unspecified ring error: {}", _0)]
+    UnspecifiedRingError(#[fail(cause)] ring::error::Unspecified),
```

Comment:
I don't recommend that people map ring's `Unspecified` error to another `Unspecified` error. Instead I would create separate errors, e.g. `KeyGenerationFailed`, `KeyAgreementFailed`, and map `Unspecified` to these based on the context.